### PR TITLE
Use the full svc cluster domain in alluxio config

### PR DESCRIPTION
**Pull request by abowloflrf**
Signed-off-by: Ruofeng Lei <ruofenglei@outlook.com>

### Ⅱ. Does this pull request fix one issue?

Use the full svc cluster domain (x.svc.cluster.local) in alluxio config.
The cluster domain is parsed from `/etc/resolv.conf`. Ref: https://stackoverflow.com/questions/52940774/kubernetes-how-to-check-current-domain-set-by-cluster-domain-from-pod

### Ⅴ. Special notes for reviews

### CODE REVIEW

**Comment by abowloflrf** [comment]
Haven't test in my environment yet. I found there is an error when creating alluxio dataset with the `v0.8.1` and the latest version (`charts/fluid/fluid` in master branch)

```
2022-10-14T18:01:48.613+0800    INFO    helm    helm/utils.go:56        coalesce.go:199: warning: destination for env is a table. Ignoring non-table value <nil>
coalesce.go:199: warning: destination for env is a table. Ignoring non-table value <nil>
Error: template: alluxio/templates/worker/statefulset.yaml:314:4: executing "alluxio/templates/worker/statefulset.yaml" at <include "alluxio.worker.tieredstoreVolumes" .>: error calling include: template: alluxio/templates/_helpers.tpl:249:17: executing "alluxio.worker.tieredstoreVolumes" at <eq .type "hostPath">: error calling eq: invalid type for comparison

2022-10-14T18:01:48.613+0800    ERROR   helm    helm/utils.go:59        failed to execute       {"args": "install -f /tmp/hbase-alluxio-values.yaml4077098121 --namespace default hbase /charts/alluxio", "error": "exit status 1"}
```

File: charts/alluxio/values.yaml [hunk 1]
```diff
@@ -443,3 +443,4 @@ runtimeIdentity:
   namespace: default
   name: xxx
 
+clusterDomain: cluster.local
```
Comment (cheyang) [inline charts/alluxio/values.yaml:446]: Please add return of line. Thanks.

**Comment by abowloflrf** [comment]
done

**Comment by abowloflrf** [comment]
> Haven't test in my environment yet. I found there is an error when creating alluxio dataset with the `v0.8.1` and the latest version (`charts/fluid/fluid` in master branch)
> 
> ```
> 2022-10-14T18:01:48.613+0800    INFO    helm    helm/utils.go:56        coalesce.go:199: warning: destination for env is a table. Ignoring non-table value <nil>
> coalesce.go:199: warning: destination for env is a table. Ignoring non-table value <nil>
> Error: template: alluxio/templates/worker/statefulset.yaml:314:4: executing "alluxio/templates/worker/statefulset.yaml" at <include "alluxio.worker.tieredstoreVolumes" .>: error calling include: template: alluxio/templates/_helpers.tpl:249:17: executing "alluxio.worker.tieredstoreVolumes" at <eq .type "hostPath">: error calling eq: invalid type for comparison
> 
> 2022-10-14T18:01:48.613+0800    ERROR   helm    helm/utils.go:59        failed to execute       {"args": "install -f /tmp/hbase-alluxio-values.yaml4077098121 --namespace default hbase /charts/alluxio", "error": "exit status 1"}
> ```

Test done, this pr works well.

Found this error because there's old version of Fluid installed before in my cluster but the old crd keeps in cluster. When I install new version the crd won't update cuz the `crd-upgrade` helm hook only excute when helm upgrade.

https://github.com/fluid-cloudnative/fluid/blob/b951c31f5a8660ceee4ab3c9bc70c77069a5f73e/charts/fluid/fluid/templates/upgrade/crd-upgrade.yaml#L8

**Review by cheyang — APPROVED** [review]

**Comment by TrafalgarZZZ** [comment]
> Is it ok to add `pre-install` to the crd-upgrade hook? @TrafalgarZZZ
I think it's ok to add `pre-install` to the hook

**Pull request merged**